subprojects/plugins/src/integTest/groovy/org/gradle/groovy/compile/BasicGroovyCompilerIntegrationSpec.groovy

```diff
@@ -77,12 +77,47 @@ abstract class BasicGroovyCompilerIntegrationSpec extends MultiVersionIntegratio
         groovyClassFile('Groovy.class').exists()
         groovyGeneratedSourceFile('Groovy$$Generated.java').exists()
         groovyClassFile('Groovy$$Generated.class').exists()
+        outputDoesNotContain('Annotation processing is not supported when incremental Groovy compilation is enabled.')
+    }
+
+    def "disableIncrementalCompilationWithAnnotationProcessor"() {
+        Assume.assumeFalse(versionLowerThan("1.7"))
+
+        when:
+        writeAnnotationProcessingBuild(
+            "", // no Java
+            "$annotationText class Groovy {}"
+        )
+        setupAnnotationProcessor()
+        enableAnnotationProcessingOfJavaStubs()
+        enableIncrementalCompilation()
+
+        then:
+        executer.expectDeprecationWarning()
+        succeeds("compileGroovy")
+        groovyClassFile('Groovy.class').exists()
+        groovyGeneratedSourceFile('Groovy$$Generated.java').exists()
+        groovyClassFile('Groovy$$Generated.class').exists()
+        outputContains(
+            ['Incremental Groovy compilation has been disabled since Java annotation processors are configured.',
```

Comment:
I think I would have concatenated the String via `+` instead of using a list and `join`.

Reply:
Done.